bin/p2-preparer/main.go

```diff
@@ -85,6 +97,13 @@ func main() {
 		logger.WithError(err).Fatalf("Could not do initial build reality at launch: %s", err)
 	}
 
+	if requiredPods != nil {
```

Comment:
You've allocated and instantiated `requiredPods` with `make()`, it cannot be `nil`. What are you trying to do with this check? You'll have to do it another way.